x-pack/plugins/task_manager/task_store.ts

```diff
@@ -103,51 +112,52 @@ export class TaskStore {
       throw new Error('TaskStore has already been initialized!');
     }
 
-    const properties = {
-      type: { type: 'keyword' },
-      task: {
-        properties: {
-          taskType: { type: 'keyword' },
-          runAt: { type: 'date' },
-          interval: { type: 'text' },
-          attempts: { type: 'integer' },
-          status: { type: 'keyword' },
-          params: { type: 'text' },
-          state: { type: 'text' },
-          user: { type: 'keyword' },
-          scope: { type: 'keyword' },
-        },
-      },
-    };
-
+    let shouldUpdate = true;
+    let existingTemplate;
     try {
-      const templateResult = await this.callCluster('indices.putTemplate', {
+      existingTemplate = await this.callCluster('indices.getTemplate', {
         name: this.index,
-        body: {
-          index_patterns: [this.index],
-          mappings: {
-            _doc: {
-              dynamic: 'strict',
-              properties,
-            },
-          },
-          settings: {
-            number_of_shards: 1,
-            auto_expand_replicas: '0-1',
-          },
-        },
       });
-      this._isInitialized = true;
-      return templateResult;
-    } catch (err) {
-      throw err;
+
+      if (existingTemplate) {
+        shouldUpdate =
+          hash(existingTemplate[this.index].mappings._doc.properties) !== hash(templateProperties);
```

Comment:
This does prevent overwriting it upon every restart, but it also means that if an older version happens to be restarted, then it will trample a newer version's template.

Reply:
I don't see this as an issue. Because index templates don't change existing indices. If you delete the index, it will then get created with the new mappings from the older templated version but before task manager would run and index a task it would update the template.

We could double check the mappings once per run before indexing or handle indexing error appropriately to determine the index was changed and doesn't match.

Reply:
I think @pickypg is right in that updates need to be controlled in that they must either install the template from scratch, upgrade it, or do nothing. If they can't upgrade it, they'll potentially break things in ways that will be hard to troubleshoot for the user: the outcome of the problem will be some kind of TypeError when an old instance of Kibana claims a new task and fields it depends on have moved.

Consider:
1. Kibana 0.1.0 starts up, installs the template from scratch
2. Kibana 1.0.0 starts up, overwrites the template
3. Kibana 0.1.0 claims a job
4. Boom.